### PR TITLE
feat: added predefined code to truncate prompt errors

### LINKED ISSUE #133: Support optional "parameters" field in function definitions
The following tool defintion must work.

```json
  {
    "tools": [
        {
            "type": "function",
            "function": {
                "name": "get_current_time",
                "description": "This function provides the current time"
            }
        }
    ]
  }
```

### PR DESCRIPTION
**Pull request by adubovik**
Blocked by https://github.com/epam/ai-dial-sdk/pull/151

Also accidentally resolves #133, because `sdk>=0.11.0` declares the tool parameters field optional, so we have to support the case when the field is missing to make the typechecker happy.

### CODE REVIEW

File: aidial_adapter_bedrock/llm/errors.py [hunk 1]
```diff
@@ -32,13 +33,9 @@ async def report_usage(self, choice: Choice) -> None:
                 stage.append_content(self.usage_message)
 
     def to_dial_exception(self) -> DialException:
-        return DialException(
-            status_code=422,
-            type="invalid_request_error",
+        return RequestValidationError(
             message=self.error_message,
             display_message=self.error_message,
-            code="invalid_argument",
```
Comment (roman-romanov-o) [inline aidial_adapter_bedrock/llm/errors.py:40]: I still don't understand, if someone actually uses it, but inside `RequestValidationError` code will be empty, so we might keep this `code="invalid_argument"` for consistency

File: aidial_adapter_bedrock/llm/errors.py [hunk 2]
```diff
@@ -61,13 +58,7 @@ def __init__(self, message: str):
...
 
     def to_dial_exception(self) -> DialException:
-        return DialException(
-            status_code=422,
-            type="invalid_request_error",
-            message=self.message,
-            code="invalid_argument",
-            param=None,
-        )
+        return RequestValidationError(message=self.message)
```
Comment (roman-romanov-o) [inline aidial_adapter_bedrock/llm/errors.py]: same question about `code="invalid_argument"` change

File: aidial_adapter_bedrock/server/exceptions.py [hunk 3]
```diff
@@ -113,13 +105,7 @@ def to_dial_exception(e: Exception) -> DialException:
...
         return e
 
-    status_code = 500
-    return DialException(
-        status_code=status_code,
-        code=str(status_code),
-        type="internal_server_error",
-        message=str(e),
-    )
+    return InternalServerError(str(e))
```
Comment (roman-romanov-o) [inline aidial_adapter_bedrock/server/exceptions.py:108]: Should we preserve `code=str(status_code)`, as we had before?

**Review by roman-romanov-o — DISMISSED** [review]

File: aidial_adapter_bedrock/llm/errors.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_bedrock/llm/errors.py:40]: fixed

File: aidial_adapter_bedrock/llm/errors.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_bedrock/llm/errors.py]: fixed

File: aidial_adapter_bedrock/server/exceptions.py (hunk 3 above)
Comment (adubovik) [reply aidial_adapter_bedrock/server/exceptions.py:108]: This is going to be populated automatically in the DIAL HTTPException constructor.

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-134.nightly-test.deltixhub.io
>E2E tests status: success

**Review by roman-romanov-o — APPROVED** [review]

**Pull request merged**